src/engine/strat_engine/backstore/metadata.rs

```diff
@@ -754,6 +785,13 @@ mod mda {
                     "MDA region data CRC".into(),
                 ));
             }
+
```

Comment:
I'm not fully understanding the entire intent of this PR, but I'm still pretty sure that this chunk of code is in the wrong place. It doesn't make sense to read the buffer that contains the variable length metadata, check its crc, and then after that decide that there was no point in reading it in the first place. I think it makes sense to move this check elsewhere.

However, I also think that this is the wrong kind of check. This is a private method, buried in a sub-module of metadata.rs. This means that it can not be invoked by any method outside the module. It is only invoked in one place in the outer module ```MDAREgions::load_state```. That method first locates the appropriate mda, of two, and only then invokes this method,```MDAHeader::load_region```. It's at the point where the newer mda is located where it makes most sense to verify if the ```STRAT_METADATA_VERSION``` is the correct one. Probably the error check and return should be moved up to ```load_state``` and ```load_region``` should contain only an assertion.

Reply:
Yep, you're right. fixed in 0ddf9edf. Do you think the check should return Ok(None) or Err()?